server/src/main/java/org/opensearch/extensions/rest/RestSendToExtensionAction.java

```diff
@@ -110,15 +117,20 @@ public RestExecuteOnExtensionResponse read(StreamInput in) throws IOException {
 
             @Override
             public void handleResponse(RestExecuteOnExtensionResponse response) {
-                responseBuilder.append(response.getResponse());
-                logger.info("Received response from extension: {}", response.getResponse());
+                logger.info("Received response from extension: {}", response.getStatus());
+                restExecuteOnExtensionResponse.setStatus(response.getStatus());
+                restExecuteOnExtensionResponse.setContentType(response.getContentType());
+                restExecuteOnExtensionResponse.setContent(response.getContent());
+                restExecuteOnExtensionResponse.setHeaders(response.getHeaders());
                 inProgressLatch.countDown();
```

Comment:
We have to figure out a way to remove latches. I think OpenSearch does it via Listeners.